examples/examples_test.go

```diff
@@ -20,181 +20,297 @@ import (
-					default:
-						contract.Assertf(false, "unknown name type: %s", res.URN.Name())
+				assertPlaintextValue := func(m map[string]interface{}, key string) {
+					assert.Truef(t, !isEncrypted(m[key]), "%s value should not encrypted", key)
+				}
+
+				for _, res := range stackInfo.Deployment.Resources {
+					if res.Type == "pulumi-nodejs:dynamic:Resource" {
+						switch res.URN.Name() {
+						case "sValue", "sApply", "cValue", "cApply":
+							assertEncryptedValue(res.Inputs, "value")
+							assertEncryptedValue(res.Outputs, "value")
+						case "pValue", "pApply":
+							assertPlaintextValue(res.Inputs, "value")
+							assertPlaintextValue(res.Outputs, "value")
+						case "pDummy":
+							assertPlaintextValue(res.Outputs, "value")
+						case "sDummy":
+							// Creation of this resource passes in a custom resource options to ensure that "value" is
+							// treated as secret.  In the state file, we'll see this as an uncrypted input with an
+							// encrypted output.
+							assertEncryptedValue(res.Outputs, "value")
+						case "rValue":
+							assertEncryptedValue(res.Inputs["value"].(map[string]interface{}), "secret")
+							assertEncryptedValue(res.Outputs["value"].(map[string]interface{}), "secret")
+							assertPlaintextValue(res.Inputs["value"].(map[string]interface{}), "plain")
+							assertPlaintextValue(res.Outputs["value"].(map[string]interface{}), "plain")
+						default:
+							contract.Assertf(false, "unknown name type: %s", res.URN.Name())
+						}
 					}
 				}
-			}
-
-			assertEncryptedValue(stackInfo.Outputs, "combinedApply")
-			assertEncryptedValue(stackInfo.Outputs, "combinedMessage")
-			assertPlaintextValue(stackInfo.Outputs, "plaintextApply")
-			assertPlaintextValue(stackInfo.Outputs, "plaintextMessage")
-			assertEncryptedValue(stackInfo.Outputs, "secretApply")
-			assertEncryptedValue(stackInfo.Outputs, "secretMessage")
-			assertEncryptedValue(stackInfo.Outputs, "richStructure")
-		},
-	})
-
-	// The compat test only works on Node 6.10.X because its uses the old 0.10.0 pulumi package, which only supported
-	// a single node version, since it had the native runtime component.
-	if nodeVer, err := getNodeVersion(); err != nil && nodeVer.Major == 6 && nodeVer.Minor == 10 {
-		examples = append(examples, integration.ProgramTestOptions{
-			Dir: path.Join(cwd, "compat/v0.10.0/minimal"),
+
+				assertEncryptedValue(stackInfo.Outputs, "combinedApply")
+				assertEncryptedValue(stackInfo.Outputs, "combinedMessage")
+				assertPlaintextValue(stackInfo.Outputs, "plaintextApply")
+				assertPlaintextValue(stackInfo.Outputs, "plaintextMessage")
+				assertEncryptedValue(stackInfo.Outputs, "secretApply")
+				assertEncryptedValue(stackInfo.Outputs, "secretMessage")
+				assertEncryptedValue(stackInfo.Outputs, "richStructure")
+			},
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func TestAccNode10CompatTests(t *testing.T) {
```

Comment:
Totally minor - but I believe these are Pulumi 0.10 tests, related to Node 6.10. Not sure why there was a relation there - but pretty sure these are not about Node 10.